filebeat/docs/how-does-filebeat-work.asciidoc

```diff
@@ -2,68 +2,68 @@
 
-In the registry file, each havested file has an entry. As files can be renamed or moved, the filename and path are not enough to identify a file. For each file unique identifiers are stored to detect if a file the same file that was harvested previously.
+The input keeps a state for each file it finds. As files can be renamed or moved, the filename and path are not enough to identify a file. For each file unique identifiers are stored to detect if a file the same file that was harvested previously.
 
 
 === Configuration options
 
 When changing the configuration options, it is important to understand what the affect of certain configuration options are. By default the filebeat config options are optimised for rotating log files and guarantee the "at least once" principle, means log lines are never lost.
 
-The configuration options will be discussed in the order `prospector -> harvester -> registrar` as this is also how they work together:
+The configuration options will be discussed in the order `inputs -> reader -> state` as this is also how they work together:
 
-==== Prospector Configuration
+==== Input Configuration
 
-As the prospector is responsible to pick up new files, the prospector must now how often to scan the file system and which files should not be picked up. The two most important config options are
+As the input module is responsible to pick up new files, the input module must know how often to scan the file system and which files should not be picked up. The two most important config options are
 
  * scan_frequency
  * ignore_older
 
 ===== scan_frequency
 
-The scan frequency defines how often a prospector scans the file system for new and existing files and checks if for each file a harvester is running (if needed). So in case a harvester is closed, it will be latest restarted again after `scan_frequency`. The default value for `scan_frequency` is 10s. This should work in most cases. Setting this value too low, leads to too much resources used from the file system as all the files are constantly scanned, setting it too high will take longer until new files are picked up. We do not recommend to set this value `<1s`.
+The scan frequency defines how often the input scans the file system for new and existing files and checks if for each file a reader is running (if needed). So in case a reader is closed, it will be latest restarted again after `scan_frequency`. The default value for `scan_frequency` is 10s. This should work in most cases. Setting this value too low, leads to too much resources used from the file system as all the files are constantly scanned, setting it too high will take longer until new files are picked up. We do not recommend to set this value `<1s`.
```

Comment:
Again, here "input" seems awkward. I won't mark all of these, but the problem is that people think of the input as being the file(s) that you're pointing to and not a component that actually does something.